org.eclipse.winery.repository/src/main/java/org/eclipse/winery/repository/backend/filebased/GitBasedRepository.java

```diff
@@ -56,30 +57,34 @@
 	public GitBasedRepository(String repositoryLocation) throws IOException, NoWorkTreeException, GitAPIException {
 		super(repositoryLocation);
 		FileRepositoryBuilder builder = new FileRepositoryBuilder();
-		Repository gitRepo = builder.setWorkTree(this.repositoryRoot.toFile()).setMustExist(false).build();
-		if (!new File(this.determineRepositoryPath(repositoryLocation) + "/.git").exists()) {
-		    gitRepo.create();
+		this.gitRepo = builder.setWorkTree(this.repositoryRoot.toFile()).setMustExist(true).build();
```

Comment:
This is just a check. If ``setMustExist`` is ``false``, the repo is not created. So afterwards, there is a check whether it is a repo and if it isn't, it will be created (call of ``create``). Afterwards, it is checked, whether the creation was successful.